### ISSUE TITLE
Incorrect golang version handling

### ISSUE THREAD

**Issue by patryk4815** [OSSIndex/vulns#129]
**Vulnerability URL**
Provide the URL to the OSS Index vulnerability. eg:
```
https://ossindex.sonatype.org/component/pkg:golang/github.com/rs/cors@v1.5.0
https://ossindex.sonatype.org/component/pkg:golang/github.com/rs/cors@1.5.0
https://ossindex.sonatype.org/vuln/f298776e-a8b6-4b91-8e51-f34b11f4c7bd

https://ossindex.sonatype.org/component/pkg:golang/github.com/hashicorp/consul
```

**Description**
What is difference between v1.5.0 and 1.5.0 in golang repos?

**Comment by ndonewar** [comment]
At the moment, the `v` in front of version numbers is required for looking up Go components in OSS Index. Any links without the `v` are technically invalid versions in OSS Index, though the component page doesn't distinguish between valid and invalid versions (it just returns vulnerabilities for any versions with known vulnerabilities).

As you may have noticed, the version ranges returned by vulnerabilities won't use the starting `v`. We may resolve that inconsistency in the future.

Whenever the OSS Index site links to any Go component pages, it should use the starting `v`. Likewise, [Nancy](https://github.com/sonatype-nexus-community/nancy) uses `v` when calling OSS Index. Have you found a link or scan tool that doesn't use the `v` for Go components? If so, that would be helpful to know. Thanks!

**Comment by patryk4815** [comment]
@ndonewar Thanks for answer!
I'm using dependabot-core and it looks like they are removing `v` prefix from `version`
https://github.com/dependabot/dependabot-core/blob/main/go_modules/lib/dependabot/go_modules/file_parser.rb#L51

BTW, This package should not be vulnerable, it has been fixed in 1.5.0:
```
https://ossindex.sonatype.org/component/pkg:golang/github.com/rs/cors@v1.5.0
https://ossindex.sonatype.org/vuln/f298776e-a8b6-4b91-8e51-f34b11f4c7bd
```

**Comment by ndonewar** [comment]
@patryk4815 My apologies, I started digging into this one more and realized I gave you incorrect information (removed above to prevent future confusion). Here's the correct info:

Go components in OSS Index should be referenced without the `v` in front of the version number. Links with a `v` are triggering a bug with no-lower-bounds ranges, which is why your example above (with a `v`) is showing that it has a vulnerability. Additionally, some (but not all) component pages are incorrectly listing versions with `v`. Both issues have been filed internally. 

As you noted, 1.5.0 should not have the issue, and that is what is shown:
https://ossindex.sonatype.org/component/pkg:golang/github.com/rs/cors@1.5.0

And lower versions (e.g. 1.4.0) should, and they do:
https://ossindex.sonatype.org/component/pkg:golang/github.com/rs/cors@1.4.0

Sorry for any inconvenience this caused!
